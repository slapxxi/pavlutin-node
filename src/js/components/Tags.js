import React from 'react';
import { Link } from 'react-router-dom';
import { withClassName } from './HOC';

function Tags({ tags, className }) {
  if (!tags) {
    return null;
  }
  return (
    <ul className={className}>
      {
        tags.map(t =>
          <li key={t} className="tags__tag">
            <Link to={`/tag/${t}`}>{t}</Link>
          </li>,
        )
      }
    </ul>
  );
}

export default withClassName('tags')(Tags);
