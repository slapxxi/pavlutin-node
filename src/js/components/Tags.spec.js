import React from 'react';
import render from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import Tags from './Tags';

const tags = [
  'test',
  'mock',
];

it('renders tags', () => {
  const tree = render.create(
    <Router>
      <Tags tags={tags} />
    </Router>,
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

it('renders tags with active tag', () => {
  const tree = render.create(
    <Router>
      <Tags tags={tags} activeTag="test" />
    </Router>,
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

it('renders tags with custom className', () => {
  const tree = render.create(
    <Router>
      <Tags tags={tags} className="test" />
    </Router>,
  );
  expect(tree.toJSON()).toMatchSnapshot();
});
