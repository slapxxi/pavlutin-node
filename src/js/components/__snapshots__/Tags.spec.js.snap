// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders tags 1`] = `
<ul
  className="tags"
>
  <li
    className="tags__tag"
  >
    <a
      href="/blog/tag/test"
      onClick={[Function]}
    >
      test
    </a>
  </li>
  <li
    className="tags__tag"
  >
    <a
      href="/blog/tag/mock"
      onClick={[Function]}
    >
      mock
    </a>
  </li>
</ul>
`;

exports[`renders tags with active tag 1`] = `
<ul
  className="tags"
>
  <li
    className="tags__tag tags__tag_active"
  >
    <a
      href="/blog/"
      onClick={[Function]}
    >
      test
    </a>
  </li>
  <li
    className="tags__tag"
  >
    <a
      href="/blog/tag/mock"
      onClick={[Function]}
    >
      mock
    </a>
  </li>
</ul>
`;

exports[`renders tags with custom className 1`] = `
<ul
  className="tags test"
>
  <li
    className="tags__tag"
  >
    <a
      href="/blog/tag/test"
      onClick={[Function]}
    >
      test
    </a>
  </li>
  <li
    className="tags__tag"
  >
    <a
      href="/blog/tag/mock"
      onClick={[Function]}
    >
      mock
    </a>
  </li>
</ul>
`;
